tests/devices/test_default_clifford.py

```diff
@@ -0,0 +1,318 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the tests for the clifford simulator based on stim
+"""
+import os
+import sys
+import math
+import pytest
+import numpy as np
+import pennylane as qml
+
+from pennylane.devices.default_clifford import _import_stim
+
+stim = pytest.importorskip("stim")
+
+pytestmark = pytest.mark.external
+
+INVSQ2 = 1 / math.sqrt(2)
+PI = math.pi
+
+
+def circuit_1():
+    """Circuit 1 with Clifford gates"""
+    qml.GlobalPhase(PI)
+    qml.CNOT(wires=[0, 1])
+    qml.PauliX(wires=[1])
+    qml.ISWAP(wires=[0, 1])
+    qml.Hadamard(wires=[0])
+    qml.WireCut(wires=[1])
+
+
+@pytest.mark.parametrize("circuit", [circuit_1])
+@pytest.mark.parametrize(
+    "expec_op",
+    [
+        qml.PauliZ(0),
+        qml.PauliZ(0) @ qml.PauliX(1),
+        qml.Hamiltonian([0.42], [qml.PauliZ(0) @ qml.PauliX(1)]),
+    ],
+)
+def test_expectation_clifford(circuit, expec_op):
+    """Test that the execution of default.clifford is possible and agrees with default.qubit"""
+    dev_c = qml.device("default.clifford")
+    dev_d = qml.device("default.qubit")
+
+    def circuit_fn():
+        circuit()
+        return qml.expval(expec_op)
+
+    qn_c = qml.QNode(circuit_fn, dev_c)
+    qn_d = qml.QNode(circuit_fn, dev_d)
+
+    assert np.allclose(qn_c(), qn_d())
+
+
+@pytest.mark.parametrize("circuit", [circuit_1])
+@pytest.mark.parametrize("state", ["tableau", "state_vector"])
+def test_state_clifford(circuit, state):
+    """Test that the execution of default.clifford is possible and agrees with default.qubit"""
+    dev_c = qml.device("default.clifford", state=state)
+    dev_d = qml.device("default.qubit")
+
+    def circuit_fn():
+        circuit()
+        return qml.state()
+
+    qn_c = qml.QNode(circuit_fn, dev_c)
+    qn_d = qml.QNode(circuit_fn, dev_d)
+
+    if state == "state_vector":
+        st1, st2 = qn_c(), qn_d()
+        phase = qml.math.divide(
+            st1, st2, out=qml.math.zeros_like(st1, dtype=complex), where=st1 != 0
+        )[qml.math.nonzero(np.round(st1, 10))]
+        if not qml.math.allclose(phase, 0.0):
+            assert qml.math.allclose(phase / phase[0], qml.math.ones(len(phase)))
```

Comment:
The states shouldn't be the same? Why this part is doing?

Reply:
It checks if there's a non-zero global phase by which states could differ.